### ISSUE TITLE
Check for omissions in types supported by both Desktop and Core

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#15255]
We have defined a common subset named .NET Standard 2.0 and analyzed (and fixed or baselined) any discrepancies between the Desktop and Core implementations of types in that subset. 

We also care about any types that Desktop and Core both have that aren't in .NET Standard 2.0. If Core is missing some members that Desktop does have, this should be a conscious decision. Example of where it wasn't: https://github.com/dotnet/corefx/pull/15238

One way to do this scan is go through https://raw.githubusercontent.com/dotnet/corefx/master/src/shims/ApiCompatBaseline.netfx461.txt, remove all the TypeMustExist lines (as these are types that Core doesn't have at all) and then manually remove all the lines relating to member exclusions we already decided to make as part of the Standard effort, as documented here https://github.com/dotnet/standard/tree/master/netstandard/ref

Anything we find we need to either add the API back or consciously decide to not do it.

### **Current status see below**

**Comment by danmosemsft** [comment]
This isn't stricly blocking release but it's important to our compat goal so it really should happen.

**Comment by danmosemsft** [comment]
@safern could you grab this one also?

**Comment by safern** [comment]
Sure! Sounds good @danmosemsft

**Comment by tarekgh** [comment]
@safern are you looking at this?

**Comment by safern** [comment]
@tarekgh not now, I'm currently focused on the netfx vertical tests.

**Comment by tarekgh** [comment]
@danmosemsft this is marked as 2.0.0 so I think it is a priority. please have @safern knows which item should be done first.

**Comment by danmosemsft** [comment]
In progress

**Comment by safern** [comment]
Thanks @danmosemsft

**Comment by danmosemsft** [comment]
- [ ] CannotMakeMemberNonVirtual
- [ ] CannotMakeMoreVisible
- [ ] CannotMakeTypeAbstract
- [ ] CannotRemoveBaseTypeOrInterface // @ViktorHofer 
- [ ] CannotSealType
- [ ] MembersMustExist // @danmosemsft
~~- [ ] TypesMustExist~~ // ignore this, non goal

**Comment by ViktorHofer** [comment]
CannotRemoveBaseTypeOrInterface analysis.

+ BaseType or Interface does exist in netstandard20 --> targets should inherit correctly from it
- BaseType or Interface does not exist in netstandard20 --> targets can't inherit

```
- System._AppDomain
- System.Runtime.InteropServices._Activator
- System.Runtime.InteropServices._Assembly
- System.Runtime.InteropServices._AssemblyBuilder
- System.Runtime.InteropServices._AssemblyName
- System.Runtime.InteropServices._Attribute
- System.Runtime.InteropServices._ConstructorBuilder
- System.Runtime.InteropServices._ConstructorInfo
- System.Runtime.InteropServices._CustomAttributeBuilder
- System.Runtime.InteropServices._EventBuilder
- System.Runtime.InteropServices._EventInfo
- System.Runtime.InteropServices._Exception
- System.Runtime.InteropServices._FieldBuilder
- System.Runtime.InteropServices._FieldInfo
- System.Runtime.InteropServices._ILGenerator
- System.Runtime.InteropServices._LocalBuilder
- System.Runtime.InteropServices._MemberInfo
- System.Runtime.InteropServices._MethodBase
- System.Runtime.InteropServices._MethodBuilder
- System.Runtime.InteropServices._MethodInfo
- System.Runtime.InteropServices._Module
- System.Runtime.InteropServices._ModuleBuilder
- System.Runtime.InteropServices._ParameterBuilder
- System.Runtime.InteropServices._ParameterInfo
- System.Runtime.InteropServices._PropertyBuilder
- System.Runtime.InteropServices._PropertyInfo
- System.Runtime.InteropServices._SignatureHelper
- System.Runtime.InteropServices._Type
- System.Runtime.Remoting.Messaging.IRemotingFormatter
- System.Xml.Serialization.SchemaImporter

+ System.Reflection.TypeInfo (System.Reflection.Emit.EnumBuilder, System.Reflection.Emit.GenericTypeParameterBuilder, System.Reflection.Emit.TypeBuilder)
+ System.Collections.ReadOnlyCollectionBase (System.Security.AccessControl.AuthorizationRuleCollection)
+ System.Data.IDataRecord (SqlDataRecord)
+ System.Collections.CollectionBase (SqlBulkCopyColumnMappingCollection)
+ System.IServiceProvider (SqlClientFactory)
+ System.ICloneable (SqlCommand, SqlConnection, SqlParameter)
```

Going to do the respective PRs now

**Comment by ViktorHofer** [comment]
Tracking changes that won't make it into netcore20 here: https://github.com/dotnet/corefx/pull/17706

**Comment by ViktorHofer** [comment]
@danmosemsft CannotSealType occurences:

```
System.Runtime.InteropServices.RuntimeEnvironment
System.Security.Cryptography.SHA1Managed
System.Security.Cryptography.SHA256Managed
System.Security.Cryptography.SHA384Managed
System.Security.Cryptography.SHA512Managed
System.Net.HttpVersion
```

All of these are also defined sealed in netstandard20. Therefore no action required.

**Comment by ViktorHofer** [comment]
@danmosemsft @weshaggard Is it intentionally that SHA*Managed, HttpVersion and RuntimeEnvironment types are sealed/static in netcoreapp and netstandard? It isn't that restricted in netfx

**Comment by danmosemsft** [comment]
@davidsh to answer whether `HttpVersion` is intentionally sealed -- breaking change from NETFX

**Comment by davidsh** [comment]
>@davidsh to answer whether HttpVersion is intentionally sealed -- breaking change from NETFX

I think @stephentoub made the change to make it sealed.  We should "un-seal" it if it is different from NETFX.

**Comment by danmosemsft** [comment]
I made that comment based on the list above/the baseline, but I'm confused now as I don't see it sealed in our code or NS20: `public static class HttpVersion`. What am I missing..

**Comment by weshaggard** [comment]
See https://github.com/dotnet/standard/issues/91 about why HttpVersion is static. I suspect the others are similar.

@danmosemsft static implies abstract sealed type in metadata.

**Comment by tarekgh** [comment]
@ViktorHofer are you still working on this one? just double checking the v2 issues are in progress.

**Comment by ViktorHofer** [comment]
Yes, this is the parent for all the child issues. Some of them are now targeting the future milestone but for some of them I'm still awaiting answers from area owners therefore they are marked with 2.0. As soon as all 2.0 child issues are closed or changed to future I will close this one.

**Comment by danmosemsft** [comment]
I am going to clsoe this one since I think those child issues adequately track it and they are basically independent.

**Comment by danmosemsft** [comment]
I'm pleased with what we accomplished with this pass.

**Comment by koromorix** [comment]
Is there a workaround for `ModuleBuilder.DefineDocument`? Does dotnet core provide any way to debug dynamically-created code?

**Comment by ViktorHofer** [comment]
> Is there a workaround for ModuleBuilder.DefineDocument? Does dotnet core provide any way to debug dynamically-created code?

Not that I'm aware of. cc @dnlharvey, @AtsushiKan
